### ISSUE TITLE
How can I find out the bash PID from a Windows cmd/powershell prompt?

### ISSUE THREAD

**Issue by rfgamaral** [Microsoft/WSL#3174]
Say I have the Windows PID from the bash process when running a WSL instance. Is it possible to get the bash process PID using cmd/powershell?

![image](https://user-images.githubusercontent.com/96476/39883650-682b3a9c-547f-11e8-8fa1-aacde44839a8.png)

How can I use the 7100 in this example, to get the 3?

**Comment by Biswa96** [comment]
Then run: `bash.exe -c "lsof" | find "bash"`

**Comment by Brian-Perkins** [comment]
There is no easy way to map between the two ID's. Currently this is mostly because we are not exposing it in any way, but beyond that there may not be a 1:1 mapping between the two due to things like [clone flags](https://github.com/Microsoft/WSL/issues/2615#issuecomment-346509481). Additionally the mapping may not be static as we try to play nice with process monitoring tools (AV, et al.) by creating new NT processes in response to exec since that is how it is normally done from Windows.

**Comment by rfgamaral** [comment]
@Biswa96 
That will never work, you might have multiple bash processes running from different windows.

@Brian-Perkins 
Ok, time for context on why I asked this, maybe I should've done that from the start...

I recently started using [Hyper](https://github.com/zeit/hyper/) and found a bunch of plugins that have problems with WSL, for instance:

- https://github.com/henrikdahl/hyper-statusline/issues/75
- https://github.com/hharnisc/hypercwd/issues/16
- https://github.com/dfrankland/hyper-tab-icons (no issue opened, but it doesn't work on WSL)

The first two have basically the same issue, they need to read the current shell absolute path from Hyper and there's no proper way to do that. The third one depends on the current foreground process running in the shell (I'm not sure if this could be fixed even if we could read the bash PID).

In other words, to fix all these plugins properly, we'd need WSL to expose the current absolute path and running process and somehow map it properly per Windows PID (because Hyper is tabbed and we can have multiple WSL instances running).

Given the current state of WSL I couldn't think of any way to get these plugins to work with WSL. Thoughts?

**Comment by fpqc** [comment]
@rfgamaral Are you running hyper over X, or are you running the Windows version? If the Windows version is compatible with cygwin, you can probably use rprichard's WSLbridge, which pipes the terminal through to a cygwin pty.

**Comment by dmex** [comment]
This ticket is also related to https://github.com/processhacker/processhacker/issues/153

We also need something to map the NT processid to the WSL processid to implement those missing features and this also applies to the same features in Task Manager and Process Explorer. 

One example is right-clicking a process in Task Manager/Process Explorer/Process Hacker and selecting the Debug menu to launch the debugger (e.g. `WslLaunchInteractive("gdb -p PID")`) but this requires the distro name and pid.

@benhillis

Are you able to follow this up?

**Comment by therealkenc** [comment]
Ref #6881

**Comment by CervEdin** [comment]
> This said, there needs to be some API to expose the mapping. `/proc/[pid]/win-pid` seems like a reasonable ask.

In mingw I believe its `/proc/[pid]/winpid
